src/tables/cmap.js

```diff
@@ -78,25 +78,95 @@ function parseCmapTableFormat4(cmap, p, data, start, offset) {
+        
+        const currentOffset = p.relativeOffset;
+
+        if ( defaultUVSOffset ) {
+            p.relativeOffset = defaultUVSOffset;
+            varSelectorRecord.defaultUVS = p.parseStruct({
+                ranges: function() {
+                    return p.parseRecordList32({
+                        startUnicodeValue: p.parseUInt24,
+                        additionalCount: p.parseByte
+                    });
+                }
+            });
+        }
+
+        if ( nonDefaultUVSOffset ) {
+            p.relativeOffset = nonDefaultUVSOffset;
+            varSelectorRecord.nonDefaultUVS = p.parseStruct({
+                uvsMappings: function() {
+                    const map = {};
+                    const list = p.parseRecordList32({
+                        unicodeValue: p.parseUInt24,
+                        glyphID: p.parseUShort
+                    });
+                    
+                    for(let i = 0; i < list.length; i += 1) {
+                        map[list[i].unicodeValue] = list[i];
+                    }
+
+                    return map;
+                }
+            });
+        }
+
+        varSelectorList[varSelector] = varSelectorRecord;
+
+        p.relativeOffset = currentOffset;
+    }
+
+    cmap.varSelectorList = varSelectorList;
+}
+
 // Parse the `cmap` table. This table stores the mappings from characters to glyphs.
-// There are many available formats, but we only support the Windows format 4 and 12.
+// There are many available formats, but we only support the Windows format 4 and 12, and format 14 as a supplement if available.
 // This function returns a `CmapEncoding` object or null if no supported format could be found.
 function parseCmapTable(data, start) {
     const cmap = {};
     cmap.version = parse.getUShort(data, start);
     check.argument(cmap.version === 0, 'cmap table version should be 0.');
 
     // The cmap table can contain many sub-tables, each with their own format.
-    // We're only interested in a "platform 0" (Unicode format) and "platform 3" (Windows format) table.
+    // We're only interested in a "platform 0" (Unicode format) and "platform 3" (Windows format) table,
+    // 
     cmap.numTables = parse.getUShort(data, start + 2);
+    let format14Parser = null;
+    let format14offset = -1;
     let offset = -1;
     for (let i = cmap.numTables - 1; i >= 0; i -= 1) {
         const platformId = parse.getUShort(data, start + 4 + (i * 8));
         const encodingId = parse.getUShort(data, start + 4 + (i * 8) + 2);
         if ((platformId === 3 && (encodingId === 0 || encodingId === 1 || encodingId === 10)) ||
             (platformId === 0 && (encodingId === 0 || encodingId === 1 || encodingId === 2 || encodingId === 3 || encodingId === 4))) {
             offset = parse.getULong(data, start + 4 + (i * 8) + 4);
-            break;
+            // allow for early break
+            if (format14Parser) {
```

Comment:
Did we mean for this to only break if format14 parser is present, because this changes the behavior from before...

Reply:
Yes, because all other cmap table formats are exclusive, we were able to break as soon as we encountered the first one that we support. But now we don't know whether a format 14 table might follow, so we have to proceed with the loop. But if we encounter a supported format and already have a table with format 14, we can indeed break to save following interations.